tests/AstTest.php

```diff
@@ -133,4 +133,19 @@ function testAstSet() {
         $this->assertSame($expected, $ast->unwrap());
         $this->assertSame($expected, $ast->array());
     }
+
+    function testAstHiddenNodes() {
+        $exposed = new Token(T_STRING, 'exposed');
+        $hidden = new Token(T_STRING, '_hidden');
+        $ast = new Ast('', [
+            'exposed' => $exposed,
+            '_hidden' => $hidden,
+            'deep' => [
+                'exposed' => $exposed,
+                '_hidden' => $hidden,
+            ]
+        ]);
+        $this->assertSame([$exposed, $exposed], $ast->tokens());
+        $this->assertSame('exposed,exposed', $ast->implode(','));
+    }
```

Comment:
@assertchris I' not sure if this change causes any harm fy, but this allows the AST to have some nodes that are not exposed when tokens are extracted to generate a `TokenStream`.